server/controllers/user.controller.js

```diff
@@ -44,37 +31,33 @@ function create(req, res, next) {
  * @returns {User}
  */
 function update(req, res, next) {
-  const user = req.user;
-  user.username = req.body.username;
-  user.mobileNumber = req.body.mobileNumber;
-
-  user.save()
-    .then(savedUser => res.json(savedUser))
-    .catch(e => next(e));
-}
-
-/**
- * Get user list.
- * @property {number} req.query.skip - Number of users to be skipped.
- * @property {number} req.query.limit - Limit number of users to be returned.
- * @returns {User[]}
- */
-function list(req, res, next) {
-  const { limit = 50, skip = 0 } = req.query;
-  User.list({ limit, skip })
-    .then(users => res.json(users))
-    .catch(e => next(e));
-}
-
-/**
- * Delete user.
- * @returns {User}
- */
-function remove(req, res, next) {
-  const user = req.user;
-  user.remove()
-    .then(deletedUser => res.json(deletedUser))
-    .catch(e => next(e));
-}
-
-export default { load, get, create, update, list, remove };
+  const user = req.userLoaded;
+  const username = req.body.username;
+  // We gotta check a few things:
+  // First we make sure we are the actual user we are modifying.
+  if(!req.user || user._id != req.user._id) {
+    let err = new APIError('Not enough  permissions to modify that user.', httpStatus.UNAUTHORIZED, true); //eslint-disable-line
+    return next(err);
+  }
+  // Next we are making sure the username doens't already exist:
+  User.findOne({ username })
+  .exec()
+  .then((_user) => {
+    if (_user && _user.id != user.id) {
+      let err = new APIError('User already exists.', httpStatus.UNAUTHORIZED, true); //eslint-disable-line
+      return next(err);
+    }
+    // Using _.pick to only get a few properties:
+    // otherwise user can set themselves to verified, etc :)
+    const pickedProps = _.pick(req.body, ['username','website','bio', 'name','email']);
+    Object.assign(user, pickedProps);
+    delete user.password;
```

Comment:
RE: add task so we can update our model to not include the password by default and instead doing it explicitly on select if we ever want it